tests/helpers/buffer-entire-stream.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Buffer an entire readable stream into a Uint8Array
+ * @param readableStream - The readable stream to buffer
+ * @returns A Uint8Array of the buffered stream
+ */
+export default async function bufferEntireStream(
+  readableStream: ReadableStream<Uint8Array>,
+): Promise<Uint8Array> {
```

Comment:
Take this with a large grain of salt, but the internet seems to think that this is a decent way to go from a `ReadableStream` to an array buffer:

```ts
const buffer = await new Response(stream).arrayBuffer();
```

Reply:
Unfortunately converting it to Response will swallow error messages from the stream, replacing them with "Failed to fetch". Webkit actually throws properly but Chromium/Firefox swallow the error message